### PR TITLE
std: generic diff and levenshtein algorithms

### PR DESCRIPTION
**Pull request by haxscramper**
- Add implementation of the *generic* myers diff and levenshtein edit
  distance algorihtms to be used in `testament` and `spellSuggest`.

Existing implementation of the diff algorithm is not generic and can be hardly made into one, due to heavily hardcoded types and unclear API, so I added new one - it might not be as optimized, but it would be much easier to reuse for writing custom diff formatters (e.g. colored)

### CODE REVIEW

File: tests/stdlib/tdiff.nim [hunk 1]
```diff
@@ -69,5 +104,96 @@ proc main() =
...
+proc main() =
+  stringAndIntDiff()
+  levenshteinDiff()
+
 main()
-static: main()
+# FIXME when compiled to js backend `formatDiffed` fails with 'cannot
+# evaluate at compile-time' error. This is not a regression, mainline works
+# the same way.
+myersDiff()
```
Comment (saem) [inline tests/stdlib/tdiff.nim:196]: Is there a simpler test case we can put into the VM tests rather than in here?

Comment (haxscramper) [reply tests/stdlib/tdiff.nim:196]: I didn't try to minimize it yet

Comment (saem) [reply tests/stdlib/tdiff.nim:196]: Not the worst, at least you described it in a comment.

**Review by saem — APPROVED** [review]
Bors r+

**Pull request merged**